### PR TITLE
For #19881 add telemetry for start on Home.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
This is using the wrong approach as we have found in https://github.com/mozilla-mobile/fenix/pull/19733. If the intent is to have this as a metric ping, you need to group it up with the metrics ping otherwise it will not be sent. You will also need to add a shared preference pref to hold onto the counter value.

Also, wondering if we went through DS for this? @rocketsroger did we have the updated docs on how to test metric pings? I think we should make sure @Amejia481 has access to that and ensure he is able to test it locally prior to the next review revision.

**Comment by Amejia481** [comment]
> This is using the wrong approach as we have found in #18711. If the intent is to have this as a metric ping, you need to group it up with the metrics ping otherwise it will not be sent. You will also need to add a shared preference pref to hold onto the counter value. See #19733 for for example.
> 
> Also, wondering if we went through DS for this? @rocketsroger did we have the updated docs on how to test metric pings? I think we should make sure @Amejia481 has access to that and ensure he is able to test it locally prior to the next review revision.

Thanks for the feedback, @rocketsroger provide me with some example and how to test glean locally, I will add wiki page for it.

File: app/metrics.yaml [hunk 1]
```diff
@@ -5445,3 +5445,35 @@ set_default_setting_experiment:
...
+      - https://github.com/mozilla-mobile/fenix/pull/19791
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2021-10-07"
+  open_tabs_tray:
+    type: counter
+    description: |
+      We know how often users open the tab traym.
```
Comment (gabrielluong) [inline app/metrics.yaml:5468]:
```suggestion
      We know how often users open the tab tray.
```

**Comment by rocketsroger** [comment]
After some conversation with the Glean team this change is probably the way to go.  Once data review is done, please add test if possible and test to make sure the metric shows up in https://debug-ping-preview.firebaseapp.com/

**Comment by gabrielluong** [comment]
> After some conversation with the Glean team this change is probably the way to go. Once data review is done, please add test if possible and test to make sure the metric shows up in https://debug-ping-preview.firebaseapp.com/

Just a note from looking at the existing code, please remember to group this up the metrics ping otherwise in `metrics.yaml`. Otherwise, it will not work. See https://github.com/mozilla-mobile/fenix/pull/19733/files

**Comment by wlach** [comment]
> This is using the wrong approach as we have found in #18711. If the intent is to have this as a metric ping, you need to group it up with the metrics ping otherwise it will not be sent. You will also need to add a shared preference pref to hold onto the counter value. See #19733 for for example.

I'm not sure about all the implementation details eithere here or in #18711, but note that you don't need to group a metric under the `metrics` category for it to be sent in the metrics ping. See #19976

**Comment by Amejia481** [comment]
@gabrielluong  and @rocketsroger, I updated the PR with the suggested changes, please let me know if I'm on the right track.

**Comment by rocketsroger** [comment]
> @gabrielluong and @rocketsroger, I updated the PR with the suggested changes, please let me know if I'm on the right track.

Have you confirmed that the metric shows up in https://debug-ping-preview.firebaseapp.com/?

**Comment by Amejia481** [comment]
> > @gabrielluong and @rocketsroger, I updated the PR with the suggested changes, please let me know if I'm on the right track.
> 
> Have you confirmed that the metric shows up in https://debug-ping-preview.firebaseapp.com/?
I'm still testing locally ^-^, but I would like to see if the changes are correct.

File: app/metrics.yaml [hunk 2]
```diff
@@ -5445,3 +5445,35 @@ set_default_setting_experiment:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2021-10-01"
+
+start_on_home:
```
Comment (gabrielluong) [inline app/metrics.yaml:5449]: I think you need to move this all under the metrics ping.

Comment (gabrielluong) [reply app/metrics.yaml:5449]: Actually scratch that from looking at https://github.com/mozilla-mobile/fenix/issues/19976. I am unsure what to do at the time of writing.

**Comment by Amejia481** [comment]
@rocketsroger  and @gabrielluong  as @wlach  mentioned, we don't need to group a metric under the metrics category for it to be sent in the `metrics` ping, I was able to send the `metrics`. 

image

**Comment by Amejia481** [comment]
I updated the PR to just move `openTabsTrayOnHomeCount.increment()` to the right event call on line `419`.

File: app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt [hunk 3]
```diff
@@ -694,6 +694,26 @@ class SettingsTest {
         assertEquals(2, settings.creditCardsSavedCount)
     }
 
+    @Test
+    fun `WHEN incrementing the enterHomeCount value THEN the value must be increased`() {
+
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt:699]: nit don't need this new line

**Comment by Amejia481** [comment]
Request for data collection review
---
1. What questions will you answer with this data?
    This data will provide statistics regarding the usage of the new start on home setting.
 
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
    This will give us an understanding of the usability of the new setting, info we will use for improving the user experience.
 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
    No alternative is available to answer this question.
 
4. Can current instrumentation answer these questions?
    No.
 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement   Description | Data Collection   Category | Tracking Bug #
-- | -- | --
Enter home screen| category 2 - Interaction data | https://github.com/mozilla-mobile/fenix/issues/19881
Open tabs tray on the home screen| category 2 - Interaction data | https://github.com/mozilla-mobile/fenix/issues/19881

6. How long will this data be collected? Choose one of the following:
    Data will be collected for 6 months initially (ending on 2021-10-07), with the option to renew at that point. 
 
7. What populations will you measure?
    No filters - all locales, channels, etc. Currently, the button is visible only in Nightly.
 
8. If this data collection is default on, what is the opt-out mechanism for users?
     Default Glean SDK opt-out mechanism.
 
9. Please provide a general description of how you will analyze this data.
    Glean / Amplitude
 
10. Where do you intend to share the results of your analysis?
      Only on Glean, Amplitude and with mobile teams.
 
11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
      No third-party tool will use this data.

**Comment by Amejia481** [comment]
cc: @tdsmith

File: app/metrics.yaml [hunk 4]
```diff
@@ -5445,3 +5445,35 @@ set_default_setting_experiment:
...
+      - metrics
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/19881
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/19791
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2021-10-07"
```
Comment (rocketsroger) [inline app/metrics.yaml:5464]: nit: the expiry is less than 4 months.  Shouldn't this at least be a year?

**Review by rocketsroger — APPROVED** [review]
Looks good.  Pending Data-review I think this is good to go.

**Comment by tdsmith** [comment]
data-review+ but I do have a drive-by data science question:

>  This will give us an understanding of the usability of the new setting, info we will use for improving the user experience.

How will you use these counters to make a decision? What decisions are you contemplating? What values would you derive from these counters that would give you an indication about whether the experience is usable?

You don't need to answer that question before proceeding but I'm not sure I understand how this should work. In the data science ticket you mention that you're interested in whether one activity closely follows another in time; events would be the natural way to instrument that.

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, in the probe definition files and the Glean probe dictionary.

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, the [Firefox telemetry opt-out](https://support.mozilla.org/en-US/kb/share-data-mozilla-help-improve-firefox).

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

n/a

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, interaction data.

5) Is the data collection request for default-on or default-off?

Default-on.

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No.

7) Is the data collection covered by the existing Firefox privacy notice?

Yes.

8) Does the data collection use a third-party collection tool?

No.

File: app/metrics.yaml (hunk 4 above)
Comment (Amejia481) [reply app/metrics.yaml]: Addressed

**Comment by Amejia481** [comment]
> data-review+ but I do have a drive-by data science question:
> 
> > This will give us an understanding of the usability of the new setting, info we will use for improving the user experience.
> 
> How will you use these counters to make a decision? What decisions are you contemplating? What values would you derive from these counters that would give you an indication about whether the experience is usable?
> 
> You don't need to answer that question before proceeding but I'm not sure I understand how this should work. In the data science ticket you mention that you're interested in whether one activity closely follows another in time; events would be the natural way to instrument that.


@rocketsroger I updated the PR to take into consideration @tdsmith  feedback, to use `events` instead of `counters`, I didn't include tests, as it doesn't worth the effort, as it's not straightforward to tests.

**Review by rocketsroger — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 5]
```diff
@@ -293,9 +293,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     }
 
     private fun startMetricsIfEnabled() {
-        if (settings().isTelemetryEnabled) {
-            components.analytics.metrics.start(MetricServiceType.Data)
-        }
+        components.analytics.metrics.start(MetricServiceType.Data)
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:296]: This is not intentional correct?

**Review by rocketsroger — CHANGES_REQUESTED** [review]
I think there's a mistake removing the settings check.

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt (hunk 5 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt:296]: Sorry, about that I missed that one,  I added it when I was testing locally

**Review by rocketsroger — APPROVED** [review]
🚢

**Pull request merged**